samples/spring-controllers/src/test/java/org/testfx/playground/view/FilteredTableTest.java

```diff
@@ -0,0 +1,82 @@
+package org.testfx.playground.view;
+
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.Scene;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyCodeCombination;
+import javafx.scene.input.KeyCombination;
+import javafx.stage.Stage;
+
+import org.junit.Test;
+import org.testfx.api.FxAssert;
+import org.testfx.framework.junit.ApplicationTest;
+import org.testfx.playground.view.FilteredTable;
+
+public class FilteredTableTest extends ApplicationTest {
+
+	private static class Item {
+		public final StringProperty name;
+		public Item(String name) {
+			this.name = new SimpleStringProperty(name);
+		}
+		public String getName() {
+			return name.get();
+		}
+	}
+
+	@Override
+	public void start(Stage stage) throws Exception {
+
+		TableView<Item> tableView = new TableView<>();
+		TableColumn<Item, String> tableColumn = new TableColumn<>("Name");
+		tableColumn.setCellValueFactory(cellData -> cellData.getValue().name);
+		tableView.getColumns().add(tableColumn);
+		tableView.setColumnResizePolicy(TableView.CONSTRAINED_RESIZE_POLICY);
+
+		FilteredTable<Item> filterableTable = new FilteredTable<>(tableView, new TextField());
+		filterableTable.setItems(toItems("foo", "bar", "baz", "Testing", "Testing, testing, 1, 2, 3"));
+		filterableTable.setFilterFunction((team, filterText) -> team.getName().contains(filterText));
+
+		stage.setScene(new Scene(filterableTable, 600, 400));
+		stage.show();
+
+	}
+
+	public ObservableList<Item> toItems(String... names) {
+		ObservableList<Item> items = FXCollections.observableArrayList();
+		for (String name : names) {
+			items.add(new Item(name));
+		}
+		return items;
+	}
+
+	@Test
+	public void testFilter() {
+
+		TableView<Item> tableView = lookup(".table-view").queryFirst();
+		TextField textField = lookup(".text-field").queryFirst();
+
+		clickOn(textField);
+
+		// add filter
+		write("Testing");
+		FxAssert.verifyThat(tableView, node -> node.getItems().size() == 2);
```

Comment:
I did not understand how to use these methods:

`FxAssert.verifyThat(String nodeQuery, Matcher<T> nodeMatcher)`
`FxAssert.verifyThat(String nodeQuery, Predicate<T> nodePredicate)`

Tried this...

`FxAssert.verifyThat(".table-view", node -> node.getItems().size() == 2);`

...but it cannot determine the type of the node.